pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/FunctionMetaDataManager.java

```diff
@@ -168,68 +151,78 @@ public synchronized boolean containsFunction(String tenant, String namespace, St
         return containsFunctionMetaData(tenant, namespace, functionName);
     }
 
-    /**
-     * Sends an update request to the FMT (Function Metadata Topic)
-     * @param functionMetaData The function metadata that needs to be updated
-     * @return a completable future of when the update has been applied
-     */
-    public synchronized CompletableFuture<RequestResult> updateFunction(FunctionMetaData functionMetaData) {
-
-        FunctionMetaData existingFunctionMetadata = null;
-        if (containsFunction(functionMetaData.getFunctionDetails().getTenant(),
-                functionMetaData.getFunctionDetails().getNamespace(),
-                functionMetaData.getFunctionDetails().getName())) {
-            existingFunctionMetadata = getFunctionMetaData(functionMetaData.getFunctionDetails().getTenant(),
-                    functionMetaData.getFunctionDetails().getNamespace(),
-                    functionMetaData.getFunctionDetails().getName());
+    public synchronized void updateFunctionOnLeader(FunctionMetaData functionMetaData, boolean delete)
```

Comment:
Currently when there are concurrent modifications, the worker will fail requests.  That doesn't seem to be done now.  All requests are accepted and 200s will be returned to the user but internally some requests will just be silently igorned.

Reply:
That behaviour hasn't changed. Users will still see 'This request is outdated, please try again' messages upon concurrent updates. That behaviour is now executed by the leader which will return that error in the /leader request.

Reply:
How is that the case?  The method "submit" is deleted. We were using the "submit()" method to put a completable future in a list and then completing the future when we were merging in the request to the local metdata cache. If there was a conflict the future is completed exceptionally with an error message.  It seems that whole workflow is removed

Reply:
I see you are throwing an IllegalArgumentException for those

Reply:
Yes. All of that workflow is gone because we no longer use submit workflow. Leader either directly updates cache or in worker mode the trailer does the update